x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -41,198 +38,156 @@ public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState license
                              AuthorizationInfo authorizationInfo) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.accessDenied(requestId, authentication, action, message, authorizationInfo);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.accessDenied(requestId, authentication, action, message, authorizationInfo);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, RestRequest request) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, request);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, request);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, String action, TransportMessage message) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, action, message);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, action, message);
         }
     }
 
     @Override
     public void tamperedRequest(String requestId, User user, String action, TransportMessage request) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.tamperedRequest(requestId, user, action, request);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.tamperedRequest(requestId, user, action, request);
         }
     }
 
     @Override
     public void connectionGranted(InetAddress inetAddress, String profile, SecurityIpFilterRule rule) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.connectionGranted(inetAddress, profile, rule);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.connectionGranted(inetAddress, profile, rule);
         }
     }
 
     @Override
     public void connectionDenied(InetAddress inetAddress, String profile, SecurityIpFilterRule rule) {
-        if (licenseState.isAuditingAllowed()) {
-            for (AuditTrail auditTrail : auditTrails) {
-                auditTrail.connectionDenied(inetAddress, profile, rule);
-            }
+        for (AuditTrail auditTrail : auditTrails) {
+            auditTrail.connectionDenied(inetAddress, profile, rule);
         }
     }
 
     @Override
     public void runAsGranted(String requestId, Authentication authentication, String action, TransportMessage message,
                              AuthorizationInfo authorizationInfo) {
-        if (licenseState.isAuditingAllowed()) {
```

Comment:
the runAs* methods are also called from the AuthorizationService, so a license check is most likely needed there as well.